src/utils/print.js

```diff
@@ -0,0 +1,47 @@
+import jQuery from 'jquery'
+/**
+ * Creates a new window (popup) with the content given in param. It will copy all the <style> tags
+ * from the app in this popup and then trigger a print request to the browser (waiting for any image
+ * to finish loading before doing so)
+ *
+ * The popup will be automatically closed after printing (also closed if the user cancels the print)
+ * so that the user may never se the little popup
+ *
+ * @param {String} htmlContent Some HTML, as String, to be added as content for the print popup
+ */
+const promptUserToPrintHtmlContent = (htmlContent) => {
+    const hiddenWindowWithContent = window.open('', '', 'height=500, width=500')
+    const document = hiddenWindowWithContent.document
+    document.write('<html><head>')
+    // grabbing all styles from main document and copying them here (to have the same styling)
+    jQuery('style').each(function () {
+        document.write(jQuery(this).clone()[0].outerHTML)
+    })
+    document.write('</head><body>')
+    document.write(htmlContent)
+    document.write('</body></html>')
+    document.close()
+
+    const printAndClose = () => {
+        hiddenWindowWithContent.print()
+        hiddenWindowWithContent.close()
+    }
+    // checking if there are images or iframe in the content
+    let pendingImageOriFrameLoading = 0
+    const images = jQuery(document).find('img,iframe')
```

Comment:
Just wondering if we know cover all cases with img and iframe ?

Reply:
I don't see another use case with our HTML popup. Usually it's a simple HTML table with a couple links at the end. Sometimes there's an image, and in some advanced cases there's an iFrame (`ch.bav.haltestellen-oev` and `ch.bfe.windenergie-geschwindigkeit_h50` are prime examples)

@rebert @ltclm Do you know any other cases of HTML popup that have content that needs loading other than `iFrame` and `img`?

Reply:
here are some more layers which are using javascript in htmlpopup (or extended tooltip):
```
cadastralwebmap.include.mako  --> ch.kantone.cadastralwebmap-farbe
cadastralwebmap_opendata.mako --> ch.swisstopo-vd.amtliche-vermessung
kernkraftwerke.mako --> ch.bfe.kernkraftwerke
kgs.mako --> ch.babs.kulturgueter
klaeranlagen.mako --> ch.bafu.gewaesserschutz-klaeranlagen_anteilq347, ch.bafu.gewaesserschutz-klaeranlagen_reinigungstyp, ch.bafu.gewaesserschutz-klaeranlagen_ausbaugroesse
lubis_bildstreifen.mako --> ch.swisstopo.lubis-bildstreifen
lubis.mako --> ch.swisstopo.lubis-luftbilder_farbe, ch.swisstopo.lubis-luftbilder_farbe, ch.swisstopo.lubis-luftbilder_schwarzweiss, ch.swisstopo.lubis-luftbilder-dritte-firmen, ch.swisstopo.lubis-luftbilder-dritte-kantone
lubis_schraegaufnahmen.mako --> ch.swisstopo.lubis-luftbilder_schraegaufnahmen
lubis_terra.mako --> ch.swisstopo.lubis-terrestrische_aufnahmen
luftfahrthindernisse.mako --> ch.bazl.luftfahrthindernis
oev_haltestellen.mako --> ch.bav.haltestellen-oev 
statistikwasserkraftanlagen.mako --> ch.bfe.statistik-wasserkraftanlagen
verkehrszaehlstellen.mako --> ch.bfe.statistik-wasserkraftanlagen
windatlas50.mako --> ch.bfe.windenergie-geschwindigkeit*
windenergieanlagen_facility.mako --> ch.bfe.windenergieanlagen
windenergieanlagen_turbine.mako --> ch.bfe.windenergieanlagen
```
since with most of these layers are loading javascript and other stuff in the extended tooltips only, there are some special layers with javascript in the default tooltip / htmlpopup:
* ch.bfe.windenergie-geschwindigkeit*
* ch.bav.haltestellen-oev

i dont think that there are more exceptions than these but i might be wrong.